tyr/servers/nginx/server.py

```diff
@@ -0,0 +1,197 @@
+# give nodes the ability to grab their own zone id with list-hosted-zones
+# Currently the following IAM Role policy is necessary:
+# ChangeResourceRecordSets on arn:aws:route53:::hostedzone/$ROUTE53_ZONE
+# This will prevent staging servers from ever updating production
+if [ $MY_ENV == "stage" ]; then
+  MY_DOMAIN=app.staghudl.com
+  ROUTE53_ZONE=Z3ETV7KVCRERYL
+elif [ $MY_ENV == "prod" ]; then
+  MY_DOMAIN=app.hudl.com
+  ROUTE53_ZONE=ZDQ066NWSBGCZ
+elif [ $MY_ENV == "test" ]; then
+  MY_DOMAIN=thorhudl.com
+  ROUTE53_ZONE=ZAH3O4H1900GY
+else
+  exit 1
+fi
+
+if [ -z "$MY_ROLE" ]
+then
+  ROLE='role[RoleBase]'
+else
+  ROLE="role[$MY_ROLE]"
+fi
+
+# Set the hostname corrently
+hostname $MY_HOSTNAME.$MY_DOMAIN
+
+# Let's add our hostname to Route 53 DNS
+mkdir '/root/route53'
+
+cat << 'EOF' > /root/route53/upsert.sh
+#!/bin/bash
+META_URL='http://169.254.169.254/latest/meta-data'
+PUBLIC_HOSTNAME=`curl -s $META_URL/public-hostname`
+
+cat << EOT > /root/route53/upsert.json
+{{
+"Comment": "Creating CNAME for $HOSTNAME",
+"Changes": [
+  {{
+    "Action": "UPSERT",
+    "ResourceRecordSet": {{
+      "Name": "$HOSTNAME.",
+      "Type": "CNAME",
+      "TTL": 60,
+      "ResourceRecords": [
+        {{
+          "Value": "$PUBLIC_HOSTNAME"
+        }}
+      ]
+    }}
+  }}
+]
+}}
+EOT
+EOF
+
+cat << EOF >> /root/route53/upsert.sh
+/usr/bin/aws route53 change-resource-record-sets --region $MY_REGION --hosted-zone-id $ROUTE53_ZONE --change-batch file:///root/route53/upsert.json >> /root/route53/upsert.log 2>&1
+EOF
+
+chmod +x /root/route53/upsert.sh
+/root/route53/upsert.sh
+# /root/route53/upsert.sh can be run at any time to generate a new .json file
+# and upsert that record into Route 53.  This may be something to run every
+# time a node is started.  Useful if a node is stopped and started and the
+# public ec2 address changes
+cat << EOF >> /etc/rc.local
+/root/route53/upsert.sh
+EOF
+
+sed -i '/requiretty/d' /etc/sudoers
+mkdir /etc/chef
+touch /etc/chef/client.rb
+echo '{validation_key}' > /etc/chef/validation.pem
+echo "
+chef_server_url '{chef_server}'
+node_name '$MY_HOSTNAME'
+environment '$MY_ENV'
+validation_client_name 'chef-validator'
+" > /etc/chef/client.rb
+curl -L https://www.opscode.com/chef/install.sh | bash;
+chef-client --server '{chef_server}' --environment $MY_ENV --node-name $MY_HOSTNAME -r "$ROLE" -L {logfile}"""
+
+        validation_key_path = os.path.expanduser('~/.chef/chef-validator.pem')
+        validation_key_file = open(validation_key_path, 'r')
+        validation_key = validation_key_file.read()
+
+        return template.format(hostname = self.hostname,
+                                fqdn = self.hostname,
+                                validation_key = validation_key,
+                                chef_server = 'http://chef.app.hudl.com/',
+                                logfile = '/var/log/chef-client.log')
+
+    def autorun(self):
```

Comment:
Since you're not modifying `autorun()` you shouldn't need to redefine it. `NginxServer` inherits from `Server` so this definition of `autorun` already exists.